server/server/src/support/mod.rs

```diff
@@ -0,0 +1,117 @@
+use actix_web::cookie::Cookie;
+use actix_web::http::header::COOKIE;
+use actix_web::web::{self};
+use actix_web::HttpRequest;
+use service::token::TokenService;
+use service::{
+    auth::{
+        validate_auth, AuthDeniedKind, AuthError, Resource, ResourceAccessRequest, ValidatedUser,
+    },
+    auth_data::AuthData,
+    service_provider::{ServiceContext, ServiceProvider},
+    settings::is_develop,
+};
+
+mod database;
+use database::get_database;
+use database::vacuum_database;
+
+const URL_PATH: &str = "/support";
+
+pub fn config_support(cfg: &mut web::ServiceConfig) {
+    cfg.route(
+        &format!("{}{}", URL_PATH, "/database"),
+        web::get().to(get_database),
+    );
+    cfg.route(
+        &format!("{}{}", URL_PATH, "/vacuum"),
+        web::post().to(vacuum_database),
+    );
+}
+
+fn validate_request(
```

Comment:
any idea why I get this?

![image](https://github.com/msupply-foundation/open-msupply/assets/9192912/4b448343-6471-4a46-915b-e933ca19ea74)

it looks like it is used..?

Reply:
oh. let me answer my own question.

```
#[cfg(not(feature = "postgres"))]
pub async fn get_database(
```

if you change to using an if statement instead, then that will fix the problem.
I'm running postgres and get the warning, because the pg version of `get_database` does not use the validate fn.